engine/src/main/java/org/terasology/engine/core/modes/StateIngame.java

```diff
@@ -101,8 +109,20 @@ public Boolean get() {
     @Override
     public void dispose(boolean shuttingDown) {
         ChunkProvider chunkProvider = context.get(ChunkProvider.class);
+        AssetTypeManager assetTypeManager = context.get(ModuleAwareAssetTypeManager.class);
         chunkProvider.dispose();
 
+        assetTypeManager.getAssetTypes().forEach(k -> {
+            for (ResourceUrn urn : k.getLoadedAssetUrns()) {
+                if (!urn.getModuleName().equals(TerasologyConstants.ENGINE_MODULE)) {
+                    k.getAsset(urn).ifPresent(Asset::dispose);
+                }
+            }
+        });
+        assetTypeManager.getAssetType(BlockFamilyDefinition.class).ifPresent(AssetType::disposeAll);
+        assetTypeManager.getAssetType(Prefab.class).ifPresent(AssetType::disposeAll);
```

Comment:
Please add a dosctring why we need to dispose these separately.

I guess this is because they are part of the engine module and are thus skipped by the statement above (https://github.com/MovingBlocks/Terasology/pull/4829/files#diff-b246f2c1e7c1a8f480631ab231b8a25b8b1b7bb5074d6b5b6ed3c6e01e44e0bcR117).

I like to have the info here about
- why do this in a separate statement
- why those two classes/how did we figure that out

Reply:
```suggestion
        // dispose engine assets that should not be kept when switching game states
        assetTypeManager.getAssetType(BlockFamilyDefinition.class).ifPresent(AssetType::disposeAll);
        assetTypeManager.getAssetType(Prefab.class).ifPresent(AssetType::disposeAll);

```